self-paced-labs/vertex-ai/vertex-ai-qwikstart/lab_exercise.ipynb

```diff
@@ -1567,158 +1717,384 @@
+    "                        default=os.environ['AIP_TEST_DATA_URI'], type=str, help='Test data GCS or BQ URI set during Vertex AI training.')\n",
+    "    # Model training args.\n",
+    "    parser.add_argument('--learning-rate', dest='learning-rate', default=0.001, type=float, help='Learning rate for optimizer.')\n",
+    "    parser.add_argument('--dropout', dest='dropout', default=0.2, type=float, help='Float percentage of DNN nodes [0,1] to drop for regularization.')    \n",
+    "    parser.add_argument('--batch-size', dest='batch-size', default=16, type=int, help='Number of examples during each training iteration.')    \n",
+    "    parser.add_argument('--n-train-examples', dest='n-train-examples', default=2638, type=int, help='Number of examples to train on.')\n",
+    "    parser.add_argument('--stop-point', dest='stop-point', default=10, type=int, help='Number of passes through the dataset during training to achieve convergence.')\n",
+    "    parser.add_argument('--n-checkpoints', dest='n-checkpoints', default=10, type=int, help='Number of model checkpoints to save during training.')\n",
+    "    \n",
+    "    args = parser.parse_args()\n",
+    "    hparams = args.__dict__\n",
     "\n",
-    "# \n",
-    "earlystopping_callback = tf.keras.callbacks.EarlyStopping(patience=2)\n",
+    "    model.train_evaluate_explain_model(hparams)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86d72c0a",
+   "metadata": {},
+   "source": [
+    "### 3. Write a `Dockerfile` for your custom ML model container"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "75e8f8af",
+   "metadata": {},
+   "source": [
+    "Third, you will write a `Dockerfile` that contains your model code as well as specifies your model code's dependencies.\n",
     "\n",
-    "# Ensure your training job's resilience to VM restarts.\n",
-    "checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(\n",
-    "    filepath=os.environ[\"AIP_MODEL_DIR\"] + '/checkpoints',\n",
-    "    save_weights_only=True,\n",
-    "    monitor='val_loss',\n",
-    "    mode='min')\n",
+    "Notice the base image below is a Vertex AI [pre-built container](https://cloud.google.com/vertex-ai/docs/training/pre-built-containers). This image includes dependencies you need for your training code such TensorFlow and BigQuery SDK which are already tested and built to save you time when building your custom container."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1456,
+   "id": "6aa86895",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Overwriting online-retail-clv-3M/Dockerfile\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%writefile {MODEL_NAME}/Dockerfile\n",
+    "# Specifies base image and tag.\n",
+    "# https://cloud.google.com/vertex-ai/docs/training/pre-built-containers\n",
+    "# us-docker.pkg.dev/vertex-ai/training/tf-cpu.2-3:latest\n",
```

Comment:
See my comment above, I believe pre-built containers are not relevant here -- or we should go all in and use pre-built containers, with no Dockerfile